src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,87 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.History;
+import duke.Storage;
+import duke.TaskList;
+import duke.ui.Ui;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+
+/**
+ * Represents a add command.
+ * The command deals with adding tasks to the list of tasks.
+ */
+public class AddCommand extends Command {
+
+    protected String type;
+    protected String description;
+
+    /**
+     * Constructs an AddCommand with specified type of task and its description.
+     *
+     * @param type The type of task.
+     * @param description The description of the task.
+     */
+    public AddCommand(String type, String description) {
+        this.type = type;
+        this.description = description;
+    }
+
+    /**
+     * Adds the task to the list of tasks and returns the relevant messages.
+     *
+     * @param tasks The TaskList that contains list of tasks.
+     * @param ui The Ui that deals with interactions with user.
+     * @param storage The Storage deals with loading and saving tasks in file.
+     * @param history The History that deals with past commands.
+     * @return The relevant messages in the form of String.
+     * @throws DukeException If description of the task is missing or in wrong format.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage, History history) throws DukeException {
```

Comment:
It is not a good practice to send UI and Storage to different classes and update them in different places. It will reduce the readability and maintainability of the code. It is better to make all the UI changes in a single place (e.g. dedicated package or class) and the same for storage. We will learn more in future lectures. For the moment keep this point in mind so that you can relate it with theories that you will learn in future